openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -701,10 +701,18 @@ def get_all_courses(cls, orgs=None, filter_=None, active_only=False, course_keys
         return course_overviews
 
     @classmethod
-    def get_all_course_keys(cls):
+    def get_all_course_keys(cls, self_paced=None):
```

Comment:
```suggestion
    def get_all_course_keys(cls, self_paced: bool | None = None):
```

We need `from __future__ import annotations` to make this work. You can remove the type from the docstring.

Let's also improve the docstring to clarify that filtering is used when this variable has a boolean value and is disabled when this value is `None`. I know it's obvious from reading the code, but let's make it clear in the docstring, too.